sqlx-postgres/src/types/geometry/path.rs

```diff
@@ -0,0 +1,372 @@
+use crate::decode::Decode;
+use crate::encode::{Encode, IsNull};
+use crate::error::BoxDynError;
+use crate::types::{PgPoint, Type};
+use crate::{PgArgumentBuffer, PgHasArrayType, PgTypeInfo, PgValueFormat, PgValueRef, Postgres};
+use sqlx_core::bytes::Buf;
+use sqlx_core::Error;
+use std::mem;
+use std::str::FromStr;
+
+const BYTE_WIDTH: usize = mem::size_of::<f64>();
+
+/// ## Postgres Geometric Path type
+///
+/// Description: Open path or Closed path (similar to polygon)
+/// Representation: Open `[(x1,y1),...]`, Closed `((x1,y1),...)`
+///
+/// Paths are represented by lists of connected points. Paths can be open, where the first and last points in the list are considered not connected, or closed, where the first and last points are considered connected.
+/// Values of type path are specified using any of the following syntaxes:
+/// ```text
+/// [ ( x1 , y1 ) , ... , ( xn , yn ) ]
+/// ( ( x1 , y1 ) , ... , ( xn , yn ) )
+///   ( x1 , y1 ) , ... , ( xn , yn )
+///   ( x1 , y1   , ... ,   xn , yn )
+///     x1 , y1   , ... ,   xn , yn
+/// ```
+/// where the points are the end points of the line segments comprising the path. Square brackets `([])` indicate an open path, while parentheses `(())` indicate a closed path.
+/// When the outermost parentheses are omitted, as in the third through fifth syntaxes, a closed path is assumed.
+///
+/// See https://www.postgresql.org/docs/16/datatype-geometric.html#DATATYPE-GEOMETRIC-PATHS
+#[derive(Debug, Clone, PartialEq)]
+pub struct PgPath {
+    pub closed: bool,
+    pub points: Vec<PgPoint>,
```

Comment:
Not sure how we feel about recycling the PgPoint type for this purpose?

Reply:
It seems reasonable to me.